Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp

```diff
@@ -2101,7 +2168,10 @@ UiProfile Spire::make_editable_table_view_profile() {
         std::make_shared<ListSingleSelectionModel>(),
         std::make_shared<ListEmptySelectionModel>()),
       [=] (const auto& table, auto row, auto column) -> QWidget* {
-        return make_row_cell(table, row, column);
+        if(row < table->get_row_size()) {
+          return make_row_cell(table, row, column);
+        }
+        return make_empty_row_cell(array_table_model, column);
```

Comment:
Explain this change please.

Reply:
It creates cell components for the last empty row.

Reply:
Why is it doing that?

Reply:
My understanding is that although the last line is a blank line, it does not mean that this line does not contain any components. Each cell in this row is the same as the cell in the previous row except that their value is null. They need to receive user input and generate a new row when submitted. This is why the function is used here.

Reply:
Why is test code implementing specifications? Test code is for testing, it's not for implementing functionality.

Reply:
Yeah, it should be a part of the EditableTableView.

Reply:
If creating empty cells was placed inside the EditableTableView, how to add a new row to the model when a cell submitted a value? Because TableModel has no push or insert interface.

Reply:
You will need to consider whatever scenarios are needed and propose a solution. I am not familiar enough with the spec or the requirements to give an answer at this time. If after thinking it over thoroughly you are unable to come up with a proposal or a solution then I can review it and provide some feedback.